packages/babel-plugin-transform-es3-reserved-words/src/index.js

```diff
@@ -0,0 +1,15 @@
+import { es3, es2017 } from "./reserved-words";
+
+export default function() {
+  const reservedKeywords = es3.filter(w => !es2017.includes(w));
+
+  return {
+    visitor: {
+      Identifier(path) {
```

Comment:
`Identifier` also covers things like `.foo`, so this would probably get called more often than we want. I'd recommend having this visit the `BindingIdentifier` and `ReferencedIdentifier` types specifically.